q2_fmt/_peds.py

```diff
@@ -466,11 +479,458 @@ def _create_masking(time_metadata, donor_df, recip_df, reference_column):
+                                               reference_column)
+
+    (metadata_df,
+     used_references) = _filter_associated_reference(reference_series,
+                                                     metadata_df, time_column,
+                                                     filter_missing_references,
+                                                     reference_column)
+
+    if len(used_references.unique()) == 1:
+        raise AssertionError("There is only one donated microbiome in your"
+                             " data. A Monte Carlo simulation shuffles"
+                             " your donated microbiome and recipient pairing"
+                             " and needs more than one donated microbiome"
+                             " to successfully shuffle.")
+    recipient = metadata_df.loc[metadata_df[reference_column].notnull()]
+
+    if len(recipient.index.unique()) == 1:
+        raise AssertionError("There is only one recipient in your"
+                             " data. A Monte Carlo simulation shuffles"
+                             " your donated microbiome and recipient pairing"
+                             " and needs more than one recipient"
+                             " to successfully shuffle.")
+
+    peds = sample_peds(
+           table=table, metadata=metadata,
+           time_column=time_column,
+           reference_column=reference_column,
+           subject_column=subject_column,
+           filter_missing_references=filter_missing_references,
+           drop_incomplete_subjects=drop_incomplete_subjects,
+           drop_incomplete_timepoint=drop_incomplete_timepoint).set_index("id")
+    actual_temp = peds["measure"]
+
+    # Mismatch simulation:
+    table = table > 0
+    recip_df = _create_recipient_table(used_references, metadata_df, table)
+    donor_df = table[table.index.isin(used_references)]
+    mismatched_df = \
+        _create_mismatched_pairs(recip_df,
+                                 metadata_df,
+                                 used_references,
+                                 reference_column)
+    duplicated_recip_table = _create_duplicated_recip_table(mismatched_df,
+                                                            recip_df)
+    donor_mask = _create_sim_masking(mismatched_df, donor_df, reference_column)
+    recip_mask = _mask_recipient(donor_mask, duplicated_recip_table)
+    # Numerator for PEDS Calc. (Number of Donor features in the Recipient)
+    num_sum = np.sum(recip_mask.values, axis=1)
+    # Denominator for PEDS Calc. (Number of unique features in the Donor)
+    donor_sum = np.sum(donor_mask, axis=1)
+    # This ignores warnings that come from dividing by 0.
+    # mismatched_peds will be Nan if the denominator is 0 and thats reasonable.
+    with warnings.catch_warnings():
+        warnings.simplefilter("ignore")
+        mismatched_peds = num_sum/donor_sum
+    mismatched_pairs_n = len(duplicated_recip_table.index)
```

Comment:
Since this value is only used inside of ``_per_subject_stats` compute it in there. (That would avoid passing an incorrect value, which would produce misleading or confusing results.)

Reply:
mistmatched_peds which we are now using to calculate mismatched_pairs_n is a list so I am no doing this calc in _per_subject_stats but need to use len(). 

my doc strings were wrong regarding this but I fixed them to reflect that mistmatched_peds is a list.